test/parallel/test-module-main-fail.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const child_process = require('child_process');
+const { promisify } = require('util');
+const execFile = promisify(child_process.execFile);
```

Comment:
Can you just use `execFileSync()` if it gets the job done. I think it would simplify the test.